synfig-studio/src/gui/workarearenderer/renderer_canvas.cpp

```diff
@@ -361,7 +361,12 @@ Renderer_Canvas::build_onion_frames()
 	current_thumb = FrameId(base_time, thumb_w, thumb_h);
 	frame_duration = Time(approximate_greater_lp(fps, 0.f) ? 1.0/(double)fps : 0.0);
 
-	// set onion_frames
+	// Store vector of Past and Future Frames to use
+	std::vector<Time> past_times, future_times;
+	// Total count of actual Past and Future Frames to use
+	int past_frames, future_frames;
```

Comment:
Now you can reduce these variable scope. They are only need inside if-statement down below.